### PR TITLE
Feature/wjiang/mapl shared

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
Make a separate MAPL_shared directory that can be shared with GMAP_pFIO  and MAPL_profiler.  All the *.h and *.H files are moved to include directory.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [ ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: MAPL_Shared/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+esma_set_this ()
+
+if(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+  if(CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 20)
+    if(CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER 17)
+      add_definitions(-D__ifort_18)
+    endif()
+  endif()
+endif()
+
```
Comment (tclune) [inline MAPL_Shared/CMakeLists.txt:10]: I don't think this logic is needed here.   That is something used with various gFTL containers.

File: MAPL_Base/CMakeLists.txt [hunk 2]
```diff
@@ -104,9 +82,9 @@ target_link_libraries (${this} PUBLIC ${ESMF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
 # in the other libaries, but this make it explicit which aspects are externally
 # used.
 file (COPY ${MAPL_SOURCE_DIR}/include/unused_dummy.H DESTINATION ${esma_include}/${this})
-file (COPY MAPL_Generic.h DESTINATION ${esma_include}/${this})
-file (COPY MAPL_Exceptions.h DESTINATION ${esma_include}/${this})
-file (COPY MAPL_ErrLog.h DESTINATION ${esma_include}/${this})
+file (COPY ${MAPL_SOURCE_DIR}/include/MAPL_Generic.h DESTINATION ${esma_include}/${this})
+file (COPY ${MAPL_SOURCE_DIR}/include/MAPL_Exceptions.h DESTINATION ${esma_include}/${this})
+file (COPY ${MAPL_SOURCE_DIR}/include/MAPL_ErrLog.h DESTINATION ${esma_include}/${this})
```
Comment (tclune) [inline MAPL_Base/CMakeLists.txt:87]: This logic should be moved into a new CMakeLists.txt in the ./include directory

File: MAPL_Shared/CMakeLists.txt (hunk 1 above)
Comment (weiyuan-jiang) [reply MAPL_Shared/CMakeLists.txt]: removed

File: MAPL_Base/CMakeLists.txt (hunk 2 above)
Comment (weiyuan-jiang) [reply MAPL_Base/CMakeLists.txt:87]: Is this for MAPL_Base to be used independently?  I am wondering how CMakeLists.txt in the ./include would know where to copy those header files

Comment (tclune) [reply MAPL_Base/CMakeLists.txt:87]: Sorry - I was rushing and not pulling all of the information together.   There should not be _any_ copy of these files.  (I was thinking this was the INSTALL step.)

${MAPL_SOURCE_DIR}/include   should be added to the target_include_directories with the BUILD_INTERFACE generator and <install>/include/MAPL for the INSTALL_INTERFACE

**Comment by weiyuan-jiang** [comment]
This PR is based on new develop branch. 1) Add more files to MAPL_Shared 2) clean up GMAO_pFIO 3) Add exception handling to MAPL_Profiler

**Comment by weiyuan-jiang** [comment]
Are those lines redundant now ?  https://github.com/GEOS-ESM/MAPL/blob/4dba4100e42c3879e1889bd2af23f02d4220edaa/GMAO_pFIO/StringVariableMap.F90#L16-L18   @tclune

**Pull request merged**